TLM/TLM/Manager/Impl/LaneConnection/LaneConnectionSubManager.cs

```diff
@@ -392,8 +393,36 @@ public class LaneConnectionSubManager :
                 }
             }
 
+            var connections = GetLaneConnections(sourceLaneId, sourceStartNode);
```

Comment:
this could be its own method right?

Reply:
what can be its own method?

Reply:
basically the whole block that was added.
than again, I can't think of a good name, probably because these are actually two different things that are happening here based on deadEnd. I'd do it the other way around.
if (deadEnd)
  removeNonDeadEndConnections()
else
  removeDwadEndConnection()
and move the laneaId loop inside.

Reply:
it would be two one line functions (unless if you count in the verbose log). Not sure this is a good idea.